src/analyze/analyze.c

```diff
@@ -2145,6 +2175,60 @@ static int do_condition(int argc, char *argv[], void *userdata) {
 }
 
 static int do_verify(int argc, char *argv[], void *userdata) {
+        int r, i = 1;
+        char **filename;
+        char *orig, *base;
+        _cleanup_free_ char *alias = NULL;
+        _cleanup_(rmrdir_and_freep) char *tempdir = NULL;
+        bool is_alias, setup_alias = false;
+
+        STRV_FOREACH(filename, strv_skip(argv, 1))
+                if (strchr(*filename, ':')) {
+                        setup_alias = true;
+                        break;
+                }
+
+        if (setup_alias) {
```

Comment:
I'd move this block into a static helper - it's general enough, and we might need a similar scratch area for future systemd-analyze use cases. Something like `static int setup_working_dir(char **ret_tempdir)` and then call it immediately after the `strchr` above